tasks/ast_tools/src/derives/mod.rs

```diff
@@ -0,0 +1,127 @@
+                        #[allow(clippy::wildcard_imports)]
+                        #use_module
+                    }
+                });
+
+                quote::quote! {
+                    #header
+
+                    #prelude
+
+                    #(#use_modules)*
+
+                    ///@@line_break
+                    #impls
+                }
+            }
+        }
+        impl $($lifetime)? $crate::codegen::Runner for $ident $($lifetime)? {
+            type Context = $crate::codegen::LateCtx;
+            type Output = $crate::derives::DeriveOutput;
+
+            fn name(&self) -> &'static str {
+                stringify!($ident)
+            }
+
+            fn run(&mut self, ctx: &$crate::codegen::LateCtx) -> $crate::Result<Self::Output> {
+                use std::collections::{HashMap, HashSet};
+                use std::vec::Vec;
+                use convert_case::{Case, Casing};
+                use itertools::Itertools;
+
+                use $crate::derives::DeriveTemplate;
+
+                let trait_name = Self::trait_name();
+                let output = ctx
+                    .schema()
+                    .into_iter()
+                    .filter(|def| def.generates_derive(trait_name))
+                    .map(|def| (def, self.derive(def, ctx)))
+                    .fold(HashMap::<&str, (HashSet<&str>, Vec<TokenStream>)>::new(), |mut acc, (def, stream)| {
+                        let module_path = def.module_path();
+                        let krate = module_path.split("::").next().unwrap();
+                        if !acc.contains_key(krate) {
+                            acc.insert(krate, Default::default());
+                        }
+                        let streams = acc.get_mut(krate).expect("We checked this right above!");
+                        streams.0.insert(module_path);
+                        streams.1.push(stream);
+                        acc
+                    })
+                    .into_iter()
+                    .sorted_by(|lhs, rhs| lhs.0.cmp(rhs.0))
+                    .fold(Vec::new(), |mut acc, (path, (modules, streams))| {
+                        let mut modules = Vec::from_iter(modules);
+                        modules.sort();
+                        acc.push((
+                            $crate::output(
+                                format!("crates/{}", path.split("::").next().unwrap()).as_str(),
+                                format!("derive_{}.rs", Self::trait_name().to_case(Case::Snake)).as_str()
+                            ),
+                            Self::template(
+                                modules,
+                                streams
+                                .into_iter()
+                                .fold(TokenStream::new(), |mut acc, it| {
+                                    acc.extend(quote::quote!{
+                                        ///@@line_break
+                                    });
```

Comment:
Do these lines mean some `///@@line_breaks` can be removed from each of the `Derive::derive` impls?

Reply:
Yes, Weirdly having it without removing those didn't result in double line breaks.